### PR TITLE
Git dependencies respect poetry.lock on install

### PR DESCRIPTION
**Pull request by wakemaster39**
# Pull Request Check List

Resolves: #2325

- [x] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

There is no documentation change required (I think)

### CODE REVIEW

**Comment by wakemaster39** [comment]
There was a missing variable being passed in preventing me from getting master running locally in the new `_execute_setup()` function. Fix is included here.

File: poetry/puzzle/provider.py [hunk 1]
```diff
@@ -815,6 +815,7 @@ def progress(self):
 
         self._in_progress = False
 
+    @classmethod
     def _execute_setup(self):
```
Comment (Jihadik) [inline poetry/puzzle/provider.py:819]: It doesn't seem you are using the class itself. In this case it is better to use `staticmethod`:
```suggestion
    @staticmethod
    def _execute_setup():
```

File: poetry/puzzle/solver.py [hunk 2]
```diff
@@ -61,24 +61,18 @@ def solve(self, use_latest=None):  # type: (...) -> List[Operation]
...
-                                pkg.source_type = "git"
-                                pkg.source_url = locked.source_url
-                                pkg.source_reference = locked.source_reference
-                                break
-
-                        if pkg_source_url != package_source_url or (
+                        lpkg = self.__find_git_locked_version(pkg)
+
+                        if lpkg:
+                            print("lpkg", lpkg.source_reference)
```
Comment (Jihadik) [inline poetry/puzzle/solver.py:67]: Are these debug prints?

File: poetry/puzzle/provider.py (hunk 1 above)
Comment (wakemaster39) [reply poetry/puzzle/provider.py:819]: this one wasn't me, I am not sure what to do here, 

https://github.com/python-poetry/poetry/blob/9f93df1140efd44f37d07fc04fe5eb9abdc105db/poetry/puzzle/provider.py#L329

Master is broken right now.

File: poetry/puzzle/solver.py (hunk 2 above)
Comment (wakemaster39) [reply poetry/puzzle/solver.py:67]: yep forgot to kil lthese

**Comment by Jihadik** [comment]
@sdispater, could we have your attention in this PR?

**Comment by 2m** [comment]
@wakemaster39 could you rebase this PR?

**Comment by abn** [comment]
@Jihadik can you rework this on top of `develop` please?

**Comment by Jihadik** [comment]
Sorry, @abn, not sure what do you mean.

**Comment by stephsamson** [comment]
@Jihadik branch off of `develop` instead of `master` 🙂

**Comment by wakemaster39** [comment]
@2m  @abn PR has been rebased and based off of development now.

**Comment by abn** [comment]
@wakemaster39 there was a broader change that would fix this issue at #2327 this might get closed in favour of that one.

**Comment by wakemaster39** [comment]
@abn thanks for getting back to me, I have adjusted this back to master. I think you meant to link a different issue though, as this is #2327 and I would be interested in seeing the broader change that would resolve this.

**Comment by abn** [comment]
@wakemaster39 yes; I meant #2720 - copy-pasta! :)

**Comment by wakemaster39** [comment]
did push yet, but this is working again once #2722 is merged

**Pull request closed without merging**